src/components/SkyBrowser/SkybrowserTabs.jsx

```diff
@@ -0,0 +1,422 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import Button from '../common/Input/Button/Button';
+import MaterialIcon from '../common/MaterialIcon/MaterialIcon';
+import ScrollOverlay from '../common/ScrollOverlay/ScrollOverlay';
+import CenteredLabel from '../common/CenteredLabel/CenteredLabel';
+import Row from '../common/Row/Row'
+import NumericInput from '../common/Input/NumericInput/NumericInput'
+import ColorPickerPopup from '../common/ColorPicker/ColorPickerPopup'
+import Checkbox from '../common/Input/Checkbox/Checkbox'
+
+import { excludeKeys } from '../../utils/helpers';
+import TooltipSkybrowser from './TooltipSkybrowser';
+import SkybrowserFocusEntry from './SkybrowserFocusEntry';
+import styles from './SkybrowserTabs.scss';
+import { Resizable } from 're-resizable';
+
+class SkybrowserTabs extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            height: 185,
+            currentHeight: 185,
+            showButtonInfo: [false, false, false, false, false],
+            showSettings: false,
+        }
+        this.createTabs = this.createTabs.bind(this);
+        this.createImageList = this.createImageList.bind(this);
+        this.onResizeStop = this.onResizeStop.bind(this);
+        this.onResize = this.onResize.bind(this);
+        this.setRef = this.setRef.bind(this);
+        this.showTooltip = this.showTooltip.bind(this);
+        this.hideTooltip = this.hideTooltip.bind(this);
+        this.createSettings = this.createSettings.bind(this);
+        this.valueToColor = this.valueToColor.bind(this);
+        this.toggleFaceCamera = this.toggleFaceCamera.bind(this);
+        this.toggleRadiusAzimuthElevation = this.toggleRadiusAzimuthElevation.bind(this);
+        this.setOpacityOfImage = this.setOpacityOfImage.bind(this);
+        this.removeImageSelection = this.removeImageSelection.bind(this);
+        this.createButtons = this.createButtons.bind(this);
+        this.toggleShowSettings = this.toggleShowSettings.bind(this);
+    }
+
+    get inheritedProps() {
+        const doNotInclude = 'closeCallback className position size title';
+        return excludeKeys(this.props, doNotInclude);
+    }
+
+    get position() {
+        if (!this.wrapper) return { top: '0px', left: '0px' };
+        const { top, right} = this.wrapper.getBoundingClientRect();
+        return { top: `${top}`, left: `${right}`};
+    }
+
+    setRef(what) {
+        return (element) => {
+          this[what] = element;
+        };
+    }
+
+    showTooltip(i) {
+    let showButtonInfoNew= [...this.state.showButtonInfo];
+    let selected = {...showButtonInfoNew[i]};
+    showButtonInfoNew[i] = true;
+    this.setState({showButtonInfoNew});
+        this.setState({showButtonInfo : showButtonInfoNew});
+    }
+
+    hideTooltip(i) {
```

Comment:
Confused about this function. Isn't `showButtonInfo` just one variable? 

Not really sure what's going on here 🤔

Reply:
showButtonInfo contains an array for all the tooltips, if they should be shown or not... And then to set an array in state in react, you need to set the whole variable.